calendar.js

```diff
@@ -19,38 +19,63 @@ function _asyncRequest(url) {
     });
 }
 
-function getCalendarDataSource() {
-    return `${baseUrl}/${year}/${month}/${day}/`;
-};
-
-async function getCalendarData() {
+async function _getCalendarData(dayOffset) {
     const allNewReleases = {}
+    const calendarDate = `${year}/${month}/${day - dayOffset}`;
+    const dataUrl = `${baseUrl}/${calendarDate}/`;
     const releaseTypeRegex = /\[([^)]+)\][ \t]+/;
     const fileTagRegex = /(.*)([\t +]\([^\]]*\))/;
     let currentPageNum = 1;
 
-    while(1) {
-        const currentPage = await _asyncRequest(`${getCalendarDataSource()}page/${currentPageNum++}`).catch(err => console.error(err));
-        if(currentPage == null) break;
-        const $ = cheerio.load(currentPage);
-        $('h2.entry-title.grid-title > a').each( (index, el) => {
-            let title = $(el).text();
-            const releaseType = releaseTypeRegex.exec(title)[1];
-            title = title.replace(releaseTypeRegex, '')
-                         .replace(fileTagRegex, '$1');
-
-            if(!allNewReleases[releaseType]) {
-                allNewReleases[releaseType] = [];
-            } 
-
-            allNewReleases[releaseType].push(title);
-        });
+    try {
```

Comment:
@mawilliamyan @Wj2ma @zuqini Is this better? :octocat:

Reply:
Dang, combined both our suggestions